packages/@aws-cdk/aws-appmesh/lib/virtual-gateway.ts

```diff
@@ -26,14 +26,9 @@ export interface IVirtualGateway extends cdk.IResource {
   readonly virtualGatewayArn: string;
 
   /**
-   * The mesh which the VirtualGateway belongs to
+   * The Mesh which the VirtualGateway belongs to
    */
   readonly mesh: IMesh;
-
-  /**
-   * Utility method to add a new GatewayRoute to the VirtualGateway
-   */
-  addGatewayRoute(id: string, route: GatewayRouteBaseProps): GatewayRoute;
```

Comment:
Why was this removed?

Reply:
My original thought was that this does not make sense to have in the `IVirtualGateway` resource. If we were to implement the inline class you suggested, we would need to also implement the `addGatewayRoute` method for that class, right? 

Thinking about it more, calling `addGatewayRoute` would be a valid call for imported virtual gateways, but you could accomplish the same thing by passing the imported gateway to the Gateway Route constructor. What are your thoughts on this?

Reply:
The way we deal with this usually in the CDK is to have an abstract module-private class called `<Resource>Base` (so in this case, `VirtualGatewayBase`) that implements the resource interface (so `IVirtualGateway` in this case), and that class contains the implementation of the methods like `addGatewayRoute`. In that case, the inline class should extend from `VirtualGatewayBase` instead of `cdk.Resource`:

```ts
  public static fromVirtualGatewayAttributes(scope: Construct, id: string, attrs: VirtualGatewayAttributes): IVirtualGateway {
    return new class extends VirtualGatewayBase {
      // ...
```

Reply:
Apparently there's already `VirtualGatewayBase`, so your work is easier: https://github.com/aws/aws-cdk/blob/8c9321b833c1dc9eed1c2e33fe3715cb0b181630/packages/%40aws-cdk/aws-appmesh/lib/virtual-gateway.ts#L75

Reply:
Looking at other classes, it looks like they implement a `ResourceBase` class in instances like this. I will do the same here